library/core/src/array/mod.rs

```diff
@@ -473,13 +550,13 @@ where
 ///
 /// If `iter.next()` panicks, all items already yielded by the iterator are
 /// dropped.
-fn collect_into_array<I, const N: usize>(iter: &mut I) -> Option<[I::Item; N]>
+fn collect_into_array<E, I, T, const N: usize>(iter: &mut I) -> Option<Result<[T; N], E>>
```

Comment:
I can't see where this function returns `None`. Am I being blind, or is the outermost `Option<_>` wrapper unnecessary?

Reply:
It's at the very end, you'll have to go further than what the github diff shows you.